### PR TITLE
Set up authorization components and tests

### PR DESCRIPTION
**Pull request by cehsu**
## Description
https://github.com/department-of-veterans-affairs/vets.gov-team/issues/12042
These changes set up the UI component of the 686 gating flow on the introduction page. The changes for the dashboard will be added in a separate PR.

## Testing done
- unit tests set up

## Screenshots
image
screen shot 2018-09-26 at 3 46 17 pm
screen shot 2018-09-26 at 3 46 32 pm
screen shot 2018-09-26 at 4 07 06 pm
screen shot 2018-09-26 at 3 44 54 pm
screen shot 2018-09-26 at 3 45 46 pm

## Acceptance criteria
- [ ] Component connected

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/disability-benefits/686/components/AuthorizationComponent.jsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    const content = (<div>
+      {isLoading && isVisible && <LoadingIndicator message="Please wait while we check your information."/>}
+      {!isLoading && isVisible && !this.props.isAuthorized && <AlertBox status="error" isVisible>
+        <AuthorizationMessage user={this.props.user}/>
+      </AlertBox>}
+      {isAuthorized && this.props.children}
+    </div>);
+
+    if (this.props.downtime) {
+      return (<DowntimeNotification
```
Comment (cehsu) [inline src/applications/disability-benefits/686/components/AuthorizationComponent.jsx]: If `evss_claims` is down, we cannot check their authorization/disability rating.

File: src/applications/disability-benefits/686/components/AuthorizationMessage.jsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      return <SystemDownView messageLine1="Sorry, our system is temporarily down while we fix a few things. Please try again later."/>;
+    }
+    if (user.profile && user.profile.status === notFound) {
+    // If va_profile is "not found", show message that we cannot find the user in our system.
+      return (
+        <SystemDownView
+          messageLine1="We couldn’t find your records with that information."
+          messageLine2="Please call the Vets.gov Help Desk at 1-855-574-7286, TTY: 1-800-877-8339. We're open Monday &#8211; Friday, 8:00 a.m. &#8211; 8:00 p.m. (ET)."/>
+      );
+    }
```
Comment (cehsu) [inline src/applications/disability-benefits/686/components/AuthorizationMessage.jsx]: This messaging and redirect logic is based on that of the [RequiredLoginView](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/platform/user/authorization/components/RequiredLoginView.jsx) component.

File: src/applications/disability-benefits/686/config/form.js [hunk 3]
```diff
@@ -18,6 +18,7 @@ import DependentField from '../components/DependentField';
 import createHouseholdMemberTitle from '../components/DisclosureTitle';
 import applicantDescription from '../../../../platform/forms/components/ApplicantDescription';
 import {
+  authorize,
```
Comment (cehsu) [inline src/applications/disability-benefits/686/config/form.js]: We will check for this on the dashboard page, and run for all gated forms that make this authorize helper available, to determine when to hide the loading indicator and reveal the saved forms/authorization errors.

File: src/applications/disability-benefits/686/components/AuthorizationMessage.jsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+const { rootUrl: increaseRootUrl } = manifest;
+import { profileStatuses } from '../helpers';
+
+const { serverError, notFound } = profileStatuses;
+const nextQuery = { next: window.location.pathname };
+const signInUrl = appendQuery('/', nextQuery);
+const verifyUrl = appendQuery('/verify', nextQuery);
+
+export default class AuthorizationMessage extends React.Component {
```
Comment (cehsu) [inline src/applications/disability-benefits/686/components/AuthorizationMessage.jsx]: This logic was split out in an effort to prepare to make the authorization logic/messaging reusable for different forms down the line.

**Comment by cehsu** [comment]
@peggygannon in addition to the form-specific error messages, several generic messages have also been included (such as when we cannot check that a user's account is verified due to a server error, or if their account couldn't be found).

**Comment by erikphansen** [comment]
In the first couple of error-state screenshots, the text is not vertically aligned with the `!` icon

File: src/applications/disability-benefits/686/components/AuthorizationMessage.jsx (hunk 2 above)
Comment (cehsu) [reply src/applications/disability-benefits/686/components/AuthorizationMessage.jsx]: @U-DON we are checking that a user is verified in order to authorize them to enter the 686 application, are these the appropriate errors to check for and error messages to display? These are based on your work on the RequiredLoginView component.

File: src/applications/disability-benefits/686/components/AuthorizationComponent.jsx [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+    return children;
+  }
+
+  render() {
+    const { isLoading, isVisible, isAuthorized } = this.props;
+
+    const content = (<div>
+      {isLoading && isVisible && <LoadingIndicator message="Please wait while we check your information."/>}
+      {!isLoading && isVisible && !this.props.isAuthorized && <AlertBox status="error" isVisible>
```
Comment (rianfowler) [inline src/applications/disability-benefits/686/components/AuthorizationComponent.jsx]: can use `!isAuthorized` here

File: src/applications/disability-benefits/686/components/AuthorizationMessage.jsx (hunk 2 above)
Comment (U-DON) [reply src/applications/disability-benefits/686/components/AuthorizationMessage.jsx]: I think those errors are quite outdated and will need to be updated.

**Review by rianfowler — APPROVED** [review]

File: src/applications/disability-benefits/686/helpers.jsx [hunk 6]
```diff
@@ -103,3 +103,13 @@ export function isVeteran(item) {
...
 
+export function authorize(isVerified) {
+  if (isVerified) {
+    // verifyDisabilityRating();
+  }
+}
+
+export const profileStatuses = { // TODO: move to platform and use in requiredLoginView
+  serverError: 'SERVER_ERROR',
+  notFound: 'NOT_FOUND'
```
Comment (erikphansen) [inline src/applications/disability-benefits/686/helpers.jsx]: I'd expect these keys to also be SNAKE_CASE just like the values are.

**Review by erikphansen — APPROVED** [review]

File: src/applications/disability-benefits/686/components/AuthorizationComponent.jsx (hunk 5 above)
Comment (cehsu) [reply src/applications/disability-benefits/686/components/AuthorizationComponent.jsx]: Thanks, updated to use `hasError` instead.

File: src/applications/disability-benefits/686/helpers.jsx (hunk 6 above)
Comment (cehsu) [reply src/applications/disability-benefits/686/helpers.jsx]: Sounds good, updated.

File: src/applications/disability-benefits/686/components/AuthorizationMessage.jsx (hunk 4 above)
Comment (cehsu) [reply src/applications/disability-benefits/686/components/AuthorizationMessage.jsx]: Thanks @U-DON, I've created an [issue](https://github.com/department-of-veterans-affairs/vets.gov-team/issues/13797) to review this content.

File: src/applications/disability-benefits/686/components/AuthorizationMessage.jsx (hunk 2 above)
Comment (cehsu) [reply src/applications/disability-benefits/686/components/AuthorizationMessage.jsx]: Thanks @U-DON, I've created an [issue](https://github.com/department-of-veterans-affairs/vets.gov-team/issues/13797) to review this content.

**Pull request merged**